### ISSUE TITLE
Nested structs leaves internal structs empty.

### ISSUE THREAD

**Issue by amany9000** [polkadot-js/api#5432]
- [ ] Bug report

Nested structs in `api.createType()` leave the internal struct empty. The internal struct is containing `Vec<u8>`.

```
    types: {
      ClaimType: {
        _enum: [
          "IssuerClaim",
          "SubjectClaim",
          "CredentialClaim"
        ]
      },
     Claim: {
        property: "Vec<u8>",
        value: "Vec<u8>",
        schema_id: "Option<i32>",
        claim_type: "ClaimType",
        issuance_requirement: "Option<Vec<IssuanceRequirement>>"
      },
     VerifiableCredentialSchema: {
	name: "Vec<u8>",
        creator: "Vec<u8>",
	issuer_claims: "Vec<Claim>",
      }
```

Claim created through createType() has the field when encoded through `.toHuman() ` but `issuer_claims` is empty in VerifiableCredentialSchema.

  - Version:
  - Environment:

    - [ ] Typescript.js
    - [ ] tsc-node

  - Language:

    - [ ] TypeScript (Version 4.9.4)

**Comment by jacogr** [comment]
What does your problematic code look like? And the logs of the data? (Simplified would help greatly as verification)

**Comment by amany9000** [comment]
```
  const claim1 = api.createType("Claim", {
    property: 'memberOf',
    value:  "PassportAuthoritySet",
    claim_type: "IssuerClaim",
  });


  const claim1Vec = api.createType("Vec<Claim>", [claim1]);
  const alice = keyring.addFromUri(`${mnemonic}//0`);


await api.tx.schemaRegistry.createSchema(
    stringToU8a("PassportIssuedVCExample"),
    stringToU8a("did:serv:5HDx7jPsiED6n47eNfERrBBRHZb59jVW6UMZZMTSBpikzvhX"),
    claim1Vec
  ).signAndSend(alice)
```

**Comment by jacogr** [comment]
A couple of issues, just looking -

1. `name: stringToU8a("Example")` this is incorrect - since it is a `Vec<u8>` and you are passing in a non-length prefixed `Uint8Array`. (When passing Uint8Arrays it needs to be in SCALE format.) In this case, just pass the string, the api will convert
2. `creator: stringToU8a("did:example:5HDx7jPsiED6n47eNfERrBBRHZb59jVW6UMZZMTSBpikzvhX"),` - Same issue as the first. It is a non length-prefixed Uint8Array value.
3. `issuedClaims` (if not used elsewhere, can just be `[{ property: 'memberOf', value:  "PassportAuthoritySet", claim_type: "IssuerClaim" }]`)`

In all cases the API will do the conversion to correct types. (As mentioned above, when passing `Uint8Array` values, it needs to be already pre-encoded in SCALE format)

**Comment by amany9000** [comment]
@jacogr when I do 
```
  const claim1 = api.createType("Claim", {
    property: stringToU8a('memberOf'),
    value:  stringToU8a("PassportAuthoritySet"),
    claim_type: "IssuerClaim",
  });
```

I get: `Struct: failed on property: Bytes:: Bytes: required length less than remainder, expected at least 6493, found 8
`

**Comment by jacogr** [comment]
Indeed - `stringToU8a('memberOf'),` would be incorrect - since it needs to be length-prefixed and it is not. See my comment around passing Uint8Arrays above - when passed it needs to be SCALE-encode, the above is not. (no compact length)

So I suggested that you don't do it

TL;DR The error is correct since the data passed doesn't make sense in SCALE (all Vec have a compact length prefix)

**Comment by amany9000** [comment]
Thank You @jacogr, scaling with .toU8a() worked.
